Source/Cake.Recipe/Content/buildProvider.cake

```diff
@@ -72,7 +72,7 @@ public enum BuildProviderType
 
 public static IBuildProvider GetBuildProvider(ICakeContext context, BuildSystem buildSystem)
 {
-    if (buildSystem.IsRunningOnAzurePipelines || buildSystem.IsRunningOnAzurePipelinesHosted)
+    if (buildSystem.IsRunningOnAzurePipelines || buildSystem.AzurePipelines.Environment.Agent.IsHosted)
```

Comment:
`IsRunningOnAzurePipelines` is enough on Cake 2.x (See https://cakebuild.net/docs/getting-started/upgrade#isrunningonazurepipelines-and-isrunningonazurepipelineshosted-properties)

```suggestion
    if (buildSystem.IsRunningOnAzurePipelines)
```